src/java/org/apache/cassandra/locator/RangesAtEndpoint.java

```diff
@@ -49,6 +49,8 @@
 
     private final InetAddressAndPort endpoint;
     private volatile Map<Range<Token>, Replica> byRange;
+    private volatile Collection<Range<Token>> fullRanges;
```

Comment:
Perhaps store the RangesAtEndpoint?  We filter(isTransient/Full) a lot, so might as well provide this feature more generally since it's just as cheap.